src/geometry/euclidean.lean

```diff
@@ -920,4 +920,328 @@ calc dist (r1 • v +ᵥ p1) (r2 • v +ᵥ p2) * dist (r1 • v +ᵥ p1) (r2 
 ... = dist p1 p2 * dist p1 p2 + (r1 - r2) * (r1 - r2) * (∥v∥ * ∥v∥)
   : by { rw [dist_eq_norm_vsub V p1, abs_mul_abs_self, mul_assoc] }
 
+/-- `p` is equidistant from two points in `s` if and only if its
+`orthogonal_projection` is. -/
+lemma dist_eq_iff_dist_orthogonal_projection_eq {s : affine_subspace ℝ P}
+    (hn : (s : set P).nonempty) (hc : is_complete (s.direction : set V)) {p1 p2 : P} (p3 : P)
+    (hp1 : p1 ∈ s) (hp2 : p2 ∈ s) :
+  dist p1 p3 = dist p2 p3 ↔
+    dist p1 (orthogonal_projection hn hc p3) = dist p2 (orthogonal_projection hn hc p3) :=
+begin
+  rw [←mul_self_inj_of_nonneg dist_nonneg dist_nonneg,
+      ←mul_self_inj_of_nonneg dist_nonneg dist_nonneg,
+      dist_square_eq_dist_orthogonal_projection_square_add_dist_orthogonal_projection_square
+        hn hc p3 hp1,
+      dist_square_eq_dist_orthogonal_projection_square_add_dist_orthogonal_projection_square
+        hn hc p3 hp2],
+  simp
+end
+
+/-- `p` is equidistant from a set of points in `s` if and only if its
+`orthogonal_projection` is. -/
+lemma dist_set_eq_iff_dist_orthogonal_projection_eq {s : affine_subspace ℝ P}
+    (hn : (s : set P).nonempty) (hc : is_complete (s.direction : set V)) {ps : set P}
+    (hps : ps ⊆ s) (p : P) :
+  (∀ p1 p2 ∈ ps, dist p1 p = dist p2 p) ↔
```

Comment:
Does it make sense to use `pairwise` for this statement?

Reply:
It looks like the only difference would be a `p1 ≠ p2` from the definition of `pairwise`, which would slightly complicate the proof of `dist_set_eq_iff_exists_dist_eq` when such a weaker condition with `p1 ≠ p2` is used to prove something without it. But maybe it would make sense to add a general lemma about `pairwise_on` for equality, which could then replace that proof (which after all has nothing to do with distance, beyond the fact that it takes values in a nonempty type).

Reply:
I've filed #3802 with the lemma about `pairwise_on` that's the essential content of `dist_set_eq_iff_exists_dist_eq`. Given that lemma, using `pairwise_on` works fine for the proofs in this PR.

Reply:
I've kicked #3802 on the merge queue :smile: